specifyweb/frontend/js_src/lib/components/WbUtils/Utils.ts

```diff
@@ -0,0 +1,443 @@
+    readonly direction: 'next' | 'previous';
+    readonly currentCellPosition?: number;
+    readonly totalCount?: number;
+    // If true and current cell is of correct type, don't navigate away
+    readonly matchCurrentCell?: boolean;
+    /*
+     * Overwrite what is considered to be a current cell
+     * Setting to [0,0] and matchCurrentCell=true allows navigation to the first
+     * cell of type (used on hitting "Enter" in the Search Box)
+     */
+    readonly currentCell?: readonly [number, number] | undefined;
+  }):
+    | readonly [
+        {
+          readonly visualRow: number;
+          readonly visualCol: number;
+        },
+        number
+      ]
+    | readonly [undefined, number] {
+    const cellMetaObject = this.workbench.cells.getCellMetaObject();
+    /*
+     * The cellMetaObject is transposed if navigation direction is "Column
+     * first".
+     * In that case, the meaning of visualRow and visualCol is swapped.
+     * resolveIndex exists to resolve the canonical visualRow/visualCol
+     */
+    const resolveIndex = (
+      visualRow: number,
+      visualCol: number,
+      first: boolean
+    ): number =>
+      (this.searchPreferences.navigation.direction === 'rowFirst') === first
+        ? visualRow
+        : visualCol;
+
+    const [currentRow, currentCol] =
+      currentCell ?? getSelectedLast(this.workbench.hot!);
+
+    const [currentTransposedRow, currentTransposedCol] = [
+      resolveIndex(currentRow, currentCol, true),
+      resolveIndex(currentRow, currentCol, false),
+    ];
+
+    const compareRows =
+      direction === 'next'
+        ? (visualRow: number) => visualRow >= currentTransposedRow
+        : (visualRow: number) => visualRow <= currentTransposedRow;
+
+    const compareCols =
+      direction === 'next'
+        ? matchCurrentCell
+          ? (visualCol: number) => visualCol >= currentTransposedCol
+          : (visualCol: number) => visualCol > currentTransposedCol
+        : matchCurrentCell
+        ? (visualCol: number) => visualCol <= currentTransposedCol
+        : (visualCol: number) => visualCol < currentTransposedCol;
+
+    let matchedCell:
+      | {
+          readonly visualRow: number;
+          readonly visualCol: number;
+        }
+      | undefined;
+    let cellIsTypeCount = 0;
+
+    const orderIt =
+      direction === 'next'
+        ? f.id
+        : <T>(array: RA<T>): RA<T> => Array.from(array).reverse();
+
+    orderIt(Object.entries(cellMetaObject)).find(([visualRowString, metaRow]) =>
+      typeof metaRow === 'object'
+        ? orderIt(Object.entries(metaRow)).find(
+            ([visualColString, metaArray]) => {
+              /*
+               * This is 10 times faster then Number.parseInt because of a slow
+               * Babel polyfill
+               */
+              const visualRow = (visualRowString as unknown as number) | 0;
+              const visualCol = (visualColString as unknown as number) | 0;
```

Comment:
hmm, the function call for fastParseInt adds a tiny bit of overhead - don't know how much though, but this code may be called thousands of times on a single user click...

ideally, we would update our babel config to not polyfil Number.parseInt at all as it has been available since 2014 in all major browsers (except for the dead ie 11)

could you please add a TODO next to `fastParseInt` like this: "update babel config to not polyfil Number.parseInt and then replace fastParseInt usages with Number.parseInt"?